controllers/registry_test.go

```diff
@@ -210,6 +210,51 @@ func TestDPAReconciler_getSecretNameAndKeyforBackupLocation(t *testing.T) {
 
 			wantProfile: "aws-provider-no-cred",
 		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fakeClient, err := getFakeClientFromObjects(tt.secret)
+			if err != nil {
+				t.Errorf("error in creating fake client, likely programmer error")
+			}
+			r := &DPAReconciler{
+				Client:        fakeClient,
+				Scheme:        fakeClient.Scheme(),
+				Log:           logr.Discard(),
+				Context:       newContextForTest(tt.name),
+				EventRecorder: record.NewFakeRecorder(10),
+			}
+
+			if tt.wantProfile == "aws-provider" {
+				tt.wantSecretKey = "cloud"
+				tt.wantSecretName = "cloud-credentials-aws"
+			}
+			if tt.wantProfile == "aws-provider-no-cred" {
+				tt.wantSecretKey = "cloud"
+				tt.wantSecretName = "cloud-credentials"
+			}
+
+			gotName, gotKey, _ := r.getSecretNameAndKey(tt.bsl.Velero.Config, tt.bsl.Velero.Credential, oadpv1alpha1.DefaultPlugin(tt.bsl.Velero.Provider))
+
+			if !reflect.DeepEqual(tt.wantSecretName, gotName) {
+				t.Errorf("expected registry container env var to be %#v, got %#v", tt.wantSecretName, gotName)
```

Comment:
message here could be something like `"expected secret name to be %#v, got %#v"`?

would also assert error is nil when calling `getSecretNameAndKey`

Reply:
Sure, will add a follow up pr to improve error messages. Thanks !